File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/Or.java
package org.uva.sea.ql.ast.expr;

public class Or extends Expr {
	Expr lhs, rhs;
	
	public Or(Expr lhs, Expr rhs) {
		this.lhs = lhs;
		this.rhs = rhs;
	}
	
	@Override
	public Boolean eval() {
		return (Boolean) lhs.eval() || (Boolean) rhs.eval();
	}
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/or.go
package binaryoperatorexpr

import "ql/ast/expr"

type Or struct {
	BinaryOperator
}

func NewOr(lhs expr.Expr, rhs expr.Expr) Or {
	return Or{BinaryOperator{Lhs: lhs, Rhs: rhs}}
}

func (o Or) Eval(s interface{}) interface{} {
	return o.Lhs.Eval(s).(bool) || o.Rhs.Eval(s).(bool)
}
